File: main/between-two-sets/between-two-sets.py
#!/usr/bin/env python3

from functools import reduce
from math import gcd

def lcm(m, n):
    return m // gcd(m, n) * n

def get_record():
    return map(int, input().split())

def get_bound(reducer):
    return reduce(reducer, sorted(get_record()))

input() # don't need n, m
lower = get_bound(lcm)
upper = get_bound(gcd)

multiples = range(lower, upper + 1, lower)
print(sum(1 for multiple in multiples if upper % multiple == 0))

File: main/between-two-sets/between-two-sets.pl
#!/usr/bin/env perl

use strict;
use warnings;
use List::Util qw(reduce);

sub gcd {
    my ($m, $n) = @_;
    return $n == 0 ? $m : gcd($n, $m % $n);
}

sub lcm {
    my ($m, $n) = @_;
    return int($m / gcd($m, $n)) * $n;
}

sub read_tokens {
    return split / /, <>;
}

sub read_as_bound (&) {
    my $reducer = \&{shift @_};
    return reduce { $reducer->($a, $b) } sort { $a <=> $b } read_tokens;
}

<>; # don't need n, m
my $lower = read_as_bound \&lcm;
my $upper = read_as_bound \&gcd;

my $count = 0;
for (my $multiple = $lower; $multiple <= $upper; $multiple += $lower) {
    if ($upper % $multiple == 0) { ++$count }
}

print "$count\n";
